src/coprocessor/dag/batch/executors/util/aggr_executor.rs

```diff
@@ -207,17 +207,15 @@ impl<Src: BatchExecutor, I: AggregationExecutorImpl<Src>> AggregationExecutor<Sr
                 .process_batch_input(&mut self.entities, src_result.data)?;
         }
 
-        if self.imp.is_partial_results_ready(src_is_drained) {
-            Ok((
-                Some(self.aggregate_partial_results(src_is_drained)?),
-                src_is_drained,
-            ))
+        let result = if self.imp.is_partial_results_ready() {
```

Comment:
should it be `if self.imp.is_partial_results_ready() || src_is_drained`?

Reply:
Updated already.